### PR TITLE
Enhancing Search

### PR DESCRIPTION
**Pull request by keatontaylor**
This PR modifies the command parser to include the keyword arguments passed in by the caller. Effectively allowing search to allow for any of the parameters defined by the ECP protocol.

Ensures that search calls can be flexible for any number of params.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 setup(
     name='roku',
-    version='3.0',
+    version='3.1',
```
Comment (iantrich) [inline setup.py]: You don't need to bump the version, I'll take care of that

**Comment by iantrich** [comment]
Can you provide me a test case I can run locally to highlight this change? Thanks.

**Comment by keatontaylor** [comment]
> Can you provide me a test case I can run locally to highlight this change? Thanks.

Sure! Something like this: 
```
from roku.core import Roku

roku = Roku('10.0.10.13')

# Search for One Punch Man matching the first result, and launch using prime video
roku.search(title="One Punch Man", provider="Prime Video", launch="true", match_any="true")

# Search for Tom Hanks with type person
roku.search(title="Tom Hanks", type="person")

# Returns search for all providers for "Voyage of the Damned via tmsid"
roku.search(keyword="voyage", tmsid="MV000058030000")
```

**Comment by keatontaylor** [comment]
@jcarbaugh looks like the CI failed, but looking at the logs this seems to be a travis-ci error not my actual code changes, is there any way to retry without submitting a change against this PR?

**Comment by iantrich** [comment]
@keatontaylor, I restarted it twice and looks to be having issues getting some images. Not going to worry about it.
Sorry, this PR slipped my mind. I'll try to test soon

**Comment by keatontaylor** [comment]
@iantrich no worries, let me know if you run into any issues.

**Comment by jcarbaugh** [comment]
@keatontaylor I just updated CI config and dependency versions... could you rebase? Thanks!

**Comment by keatontaylor** [comment]
Updated.

**Pull request merged**